### ISSUE TITLE
Using keyVaultCodesignCertificateName doesn't appear to work.

### ISSUE THREAD

**Issue by BMI-Mike** [microsoft/AL-Go#595]
I've tried using the main and preview releases of AL-Go-AppSource to get it working with an HSM-backed Azure Key Vault with a new code-signing certificate.  All is well if I use the prior CodeSignCertificateUrl and CodeSignCertificatePassword values with an older non-HSM code-signing certificate (pfx file).

In the logs, I never see AL-Go try to read this new keyVaultCodesignCertificateName value out of the Key Vault nor from GitHub secrets.

**Comment by aholstrup1** [comment]
What do you see in the logs under the Sign action? 

Do you have the keyVaultCodesignCertificateName setting in your Al-Go settings? And the Azure_Credentials secret?

**Comment by BMI-Mike** [comment]
Adding keyVaultCodesignCertificateName to the settings file appeared to get me further along, but now I am encountering this error:
<img width="704" alt="image" src="https://github.com/microsoft/AL-Go/assets/35507227/329ea859-9300-4617-90e1-7fb0bdf53a04">

**Comment by aholstrup1** [comment]
Thanks for sharing! Indeed the keyVaultCodesignCertificateName is a setting that should be in the al-go settings files. 

The issue you're encountering with the Sign action is a bug on our side. I've opened up a PR that addresses this bug

**Comment by aholstrup1** [comment]
@BMI-Mike - The fix is in and has been deployed to preview. If you try again now you shouldn't see the same error

**Comment by freddydk** [comment]
> @BMI-Mike - The fix is in and has been deployed to preview. If you try again now you shouldn't see that error anymore

You will need to run Update AL-Go System files first (as the fix is in _BuildALGoProject.yaml)

**Comment by BMI-Mike** [comment]
@aholstrup1 @freddydk Thank you both!  I'll give this a try shortly.

**Comment by BMI-Mike** [comment]
It took me a little while to sort out RBAC permissions on the Certificate stored in the Key Vault, but I am happy to confirm that code signing has been completed successfully!

I had to add the Key Vault Certificates Officer role to App Registration directly on the Certificate within the Key Vault.  Does this sound appropriate?

**Comment by aholstrup1** [comment]
Thanks for confirming that! Glad to hear it's working :) 

When I've used the sign action, I've opted to give it the fewest permission I could get away with. So I created an access policy  and assigned the app registration the following permissions: 
_Certificate permission: Get_ 
_Cryptographic Operations: Sign_

Is your App Registration only assigned permissions through the "Key Vault Certificates Officer" role? Looking at the [documentation](https://learn.microsoft.com/en-us/azure/role-based-access-control/built-in-roles#key-vault-certificates-officer) it doesn't look like it has the Sign permission.
